From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: chickeneer <emcchickeneer@gmail.com>
Date: Sat, 29 Nov 2025 13:01:51 -0600
Subject: [PATCH] Add a ShelfConnectionEvent for cancelling

Goal is to be able to cancel certain connections from occurring
if the blocks on the left or right are incompatible

diff --git a/src/main/java/com/empireminecraft/empirecraft/event/block/CheckShelfConnectionEvent.java b/src/main/java/com/empireminecraft/empirecraft/event/block/CheckShelfConnectionEvent.java
new file mode 100644
index 0000000000000000000000000000000000000000..929fc931d1baeb702133d7fce8d229655aa3257c
--- /dev/null
+++ b/src/main/java/com/empireminecraft/empirecraft/event/block/CheckShelfConnectionEvent.java
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2025. Starlis LLC / dba Empire Minecraft
+ *
+ * This source code is proprietary software and must not be redistributed without Starlis LLC's approval
+ */
+
+package com.empireminecraft.empirecraft.event.block;
+
+import org.bukkit.block.Block;
+import org.bukkit.event.Cancellable;
+import org.bukkit.event.HandlerList;
+import org.bukkit.event.block.BlockEvent;
+import org.jspecify.annotations.NullMarked;
+
+@NullMarked
+public class CheckShelfConnectionEvent extends BlockEvent implements Cancellable {
+    private final Block connectTo;
+    private boolean cancelled;
+
+    public CheckShelfConnectionEvent(Block block, Block connectTo) {
+        super(block);
+        this.connectTo = connectTo;
+    }
+
+    public Block getConnectTo() {
+        return connectTo;
+    }
+
+    @Override
+    public boolean isCancelled() {
+        return cancelled;
+    }
+
+    @Override
+    public void setCancelled(boolean cancelled) {
+        this.cancelled = cancelled;
+    }
+
+    private static final HandlerList handlers = new HandlerList();
+
+    public HandlerList getHandlers() {
+        return handlers;
+    }
+
+    public static HandlerList getHandlerList() {
+        return handlers;
+    }
+}
